scripts/backout_data_collection.py

```diff
@@ -0,0 +1,169 @@
+import json
+import logging
+import os
+import sys
+
+from tqdm import tqdm
+
+sys.path.append("../bugbug")
+from bugbug import bugzilla, db, repository
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def download_databases() -> None:
+    logger.info("Downloading bugs database...")
+    assert db.download(bugzilla.BUGS_DB)
+
+    logger.info("Downloading commits database...")
+    assert db.download(repository.COMMITS_DB, support_files_too=True)
+
+
+def preprocess_commits_and_bugs():
+    logger.info("Preprocessing commits")
+    commit_dict = {}
+    bug_to_commit_dict = {}
+
+    # store commits with their hashes and bug IDs as keys
+    for commit in tqdm(
+        repository.get_commits(
+            include_no_bug=True, include_backouts=True, include_ignored=True
+        ),
+        desc="Processing commits",
+    ):
+        commit_dict[commit["node"]] = commit
+
+        if commit["bug_id"] not in bug_to_commit_dict:
+            bug_to_commit_dict[commit["bug_id"]] = [commit]
+        else:
+            bug_to_commit_dict[commit["bug_id"]].append(commit)
+
+    logger.info("Preprocessing bugs")
+    bug_dict = {}
+
+    num_lines = sum(1 for line in open(bugzilla.BUGS_DB, "r"))
+
+    # store bugs with their bug IDs as keys
+    with open(bugzilla.BUGS_DB, "r") as f:
+        for line in tqdm(f, total=num_lines, desc="Processing bugs"):
+            bug = json.loads(line)
+            bug_dict[bug.get("id")] = bug
+
+    return commit_dict, bug_to_commit_dict, bug_dict
+
+
+def is_bug_fixed(bug_id: int, bug_dict: dict) -> bool:
+    bug = bug_dict.get(bug_id)
+
+    if bug:
+        return bug.get("resolution") == "FIXED"
+
+    return False
+
+
+def filter_commits(
+    directory_path: str,
+    destination_filepath: str,
+    count_limit: int,
+    bug_dict: dict,
+    commit_dict: dict,
+    bug_to_commit_dict: dict,
+) -> None:
+    filtered_list = []
+    counter = 0
+
+    pbar = tqdm(total=count_limit, desc="Filtering commits")
+
+    for commit in repository.get_commits(
+        include_no_bug=False, include_backouts=True, include_ignored=False
+    ):
+        # add commit if it was backed out and the bug is fixed
+        if len(commit["backedoutby"]) > 0 and is_bug_fixed(
+            bug_id=commit["bug_id"], bug_dict=bug_dict
+        ):
```

Comment:
We could try simplifying this:
```suggestion
        bug = bug_dict[commit["bug_id"]]
        if commit["backedoutby"] and bug["resolution"] == "FIXED":
```